src/test/scala/org/ergoplatform/nodeView/mempool/ErgoMemPoolSpec.scala

```diff
@@ -37,24 +37,37 @@ class ErgoMemPoolSpec extends FlatSpec
       pool = pool.putWithoutCheck(Seq(tx))
     }
     txs.foreach { tx =>
-      pool.putIfValid(tx, us)._2 shouldBe ProcessingOutcome.Declined
+      pool.process(tx, us)._2.isInstanceOf[ProcessingOutcome.Declined] shouldBe true
     }
   }
 
   it should "decline transactions invalidated earlier" in {
     val us = createUtxoState()._1
     var pool = ErgoMemPool.empty(settings)
     forAll(invalidBlockTransactionsGen) { blockTransactions =>
-      blockTransactions.txs.foreach(tx => pool = pool.putIfValid(tx, us)._1)
-      blockTransactions.txs.foreach(tx => pool.putIfValid(tx, us)._2 shouldBe ProcessingOutcome.Declined)
+      blockTransactions.txs.foreach(tx => pool = pool.process(tx, us)._1)
+      blockTransactions.txs.foreach(tx =>
+        pool.process(tx, us)._2.isInstanceOf[ProcessingOutcome.Declined] shouldBe true)
+    }
+  }
+
+  it should "decline transactions not meeting min fee" in {
+    val (us, bh) = createUtxoState()
+    val genesis = validFullBlock(None, us, bh, Random)
+    val wus = WrappedUtxoState(us, bh, stateConstants).applyModifier(genesis).get
+    val txs = validTransactionsFromUtxoState(wus, Random)
+    val maxSettings = settings.copy(nodeSettings = settings.nodeSettings.copy(minimalFeeAmount = Long.MaxValue))
```

Comment:
Also check that the same transaction will be accepted if `minimalFeeAmount = 0`

Reply:
Done